lib/FieldType/Mapper/DateFormMapper.php

```diff
@@ -59,6 +60,7 @@ public function mapFieldValueForm(FormInterface $fieldForm, FieldData $data)
                         [
                             'required' => $fieldDefinition->isRequired,
                             'label' => $fieldDefinition->getName(),
+                            'attr' => ['data-timezone_offset' => $timezoneOffsetSeconds]
```

Comment:
data attributes should be with `-` not with `_`. Thanks to that in `dataset` you have the camelCase properties

Reply:
Added `-` as requested.